source/java/src/main/java/com/uber/neuropod/NeuropodTensor.java

```diff
@@ -0,0 +1,131 @@
+package com.uber.neuropod;
+
+import java.util.List;
+
+/**
+ * The interface Neuropod tensor, which stores the data of a tensor. NeuropodTensors are backed
+ * by a buffer, and the buffer may be allocated by the C++ side(inference result) or the Java side.
+ */
+interface NeuropodTensor {
```

Comment:
As I mentioned above, I think we should combine `NeuropodJavaTensor` and `NeuropodNativeTensor` and turn this interface into a class.